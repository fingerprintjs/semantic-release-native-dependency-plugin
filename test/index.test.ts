import { describe, expect, it } from '@jest/globals'
import PluginConfig from '../src/@types/pluginConfig'
import { GenerateNotesContext } from 'semantic-release'

const { generateNotes } = require('../src/index')

const cwd = process.cwd()
const pluginConfig = {
  androidPath: 'android',
  androidGradleTaskName: 'print',
  iOSPodSpecJsonPath: 'podspec.json',
} satisfies PluginConfig
const generateNotesContext = {
  cwd: cwd,
} satisfies Pick<GenerateNotesContext, 'cwd'>

describe('index', () => {
  describe('generateNotes', () => {
    it('is a function', () => {
      expect(generateNotes).toBeInstanceOf(Function)
    })
    it('throws error on cwd not set', async () => {
      const testGenerateNotesContext = {
        ...generateNotesContext,
        cwd: undefined,
      }
      await expect(generateNotes(pluginConfig, testGenerateNotesContext)).rejects.toThrowErrorMatchingSnapshot('noCwd')
    })
    it('throws error on androidGradleTaskName not set', () => {
      const testPluginConfig = {
        ...pluginConfig,
        androidGradleTaskName: undefined,
      }

      expect(generateNotes(testPluginConfig, generateNotesContext)).rejects.toThrowErrorMatchingSnapshot(
        'noAndroidGradleTaskName'
      )
    })
  })
})
