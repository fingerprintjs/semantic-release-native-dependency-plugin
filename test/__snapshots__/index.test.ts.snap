// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index generateNotes throws error on androidGradleTaskName not set: noAndroidGradleTaskName 1`] = `"Android gradle task name should be defined."`;

exports[`index generateNotes throws error on cwd not set: noCwd 1`] = `"Current working directory is required to detect native SDK versions."`;
